### PR TITLE
allowing wildcard imports AND requiring api def

### PR DESCRIPTION
**Pull request by sfletche**
**WHAT I PROPOSE**
Allowing wildcard imports 
Encouraging clearly defined APIs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1089,22 +1089,34 @@ Other Style Guides
...
+    // good named exports
+    const foo = 23;
+    const bar = 42;
+    export { foo, bar };
+
+    // bad default export
+    export default function foo() { ... }
+    ...
+
+    // good defualt export
```
Comment (manjumuthaiya) [inline README.md]: typo on 'default'

Comment (sfletche) [reply README.md]: thanks! 
done!

**Comment by lancempoe** [comment]
do you mind putting a link to es6 and removing the note about es5?

File: README.md [hunk 2]
```diff
@@ -1089,22 +1089,34 @@ Other Style Guides
     export default es6;
     ```
 
-  <a name="modules--no-wildcard"></a><a name="10.2"></a>
-  - [10.2](#modules--no-wildcard) Do not use wildcard imports.
-
-    > Why? This makes sure you have a single default export,
-    > and forces you to define the API (keeping things organized)
+  <a name="modules--define-api"></a><a name="10.2"></a>
+  - [10.2](#modules--define-api) Always define your module's API at the bottom of the file.
```
Comment (lancempoe) [inline README.md:1093]: how do you feel about saying always if you have more than 1? Having this when you only have 1 will also fource you to override the eslint as you shouldn't have a export { somthing } when you only have method. See selectors/index.js as an example

**Comment by lancempoe** [comment]
small thought 👍

File: README.md (hunk 2 above)
Comment (sfletche) [reply README.md:1093]: Ah.  In most cases when you are exporting only 1 thing (like a class or a function), it should be a default export.

In `selectors/index.js` this is a special case in which we expect that future work will warrant the use of the exported object list and including that syntax now allows for minimal impact when future items are added.

**Pull request merged**